pkg/reconciler/build/build.go

```diff
@@ -107,8 +105,13 @@ func NewController(
 		UpdateFunc: controller.PassNew(impl.Enqueue),
 	})
 
-	// TODO(mattmoor): Set up a Pod informer, so that Pod updates
-	// trigger Build reconciliations.
+	// Set up a Pod informer, so that Pod updates trigger Build
+	// reconciliations.
+	kubeinformers.Core().V1().Pods().Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    r.addPodEvent,
+		UpdateFunc: r.updatePodEvent,
+		DeleteFunc: r.deletePodEvent,
```

Comment:
For this to be properly level-based, these should use `impl.EnqueueControllerOf` (possibly wrapped with `controller.PassNew`).  For a Pod informer, you also certainly want to use a Filter on the OwnerReference Kind being Build.

In level-based reconciliation, all of the logic is oriented in terms of reconciling Build keys (`Reconcile()`), and these triggers are very thin, they just figure out what Build to queue and enqueue it.

Reply:
Thanks for that tip. I tried switching to this and had some trouble in tests getting a signal that the enqueued update happened. Do you happen to have a link to how you handle this in Serving?

Also, good idea filtering for build-owned pods. How do I express that filter to `podsLister`? [`NewSharedInformerFactoryWithOptions`](https://godoc.org/k8s.io/client-go/informers#NewSharedInformerFactoryWithOptions) with a [`TweakListOptions`](https://godoc.org/k8s.io/client-go/informers#WithTweakListOptions) to specify a `FieldSelector`?